code/modules/mob/living/simple_animal/hostile/grue.dm

```diff
@@ -0,0 +1,591 @@
+		 						a_31,a_32,a_33,a_34,
+			 					a_41,a_42,a_43,a_44,
+			 					a_51,a_52,a_53,a_54)
+//		else if(b_matrix_testing_override) //not used for now
+//			colourmatrix = list(b_11, b_12, b_22,
+//						 b_21,b_22, b_23,
+//						 b_31, b_32, b_33,
+//						 b_41, b_42, b_43)
+
+/mob/living/simple_animal/hostile/grue/Stat()
+	..()
+	if(statpanel("Status"))
+//		stat(null, "Intent: [a_intent]")
+//		stat(null, "Move Mode: [m_intent]")
+		if(ismoulting)
+			stat(null, "Moulting progress: [round(100*(1-moulttimer/moulttime),0.1)]%")
+		if(lifestage<3) //not needed for adults
+			stat(null, "Shadow power: [round(shadowpower,0.1)]/[round(maxshadowpower,0.1)]")
+		if(lifestage>=3)
+			if(config.grue_egglaying)
+				stat(null, "Reproductive energy: [eatencharge]")
+			stat(null, "Sentient life forms eaten: [eatencount]")
+
+/mob/living/simple_animal/hostile/grue/gruespawn
+	lifestage=1
+
+/mob/living/simple_animal/hostile/grue/grueling
+	lifestage=2
+
+//Moulting into more mature forms.
+/mob/living/simple_animal/hostile/grue/verb/moult()
+	set name = "Moult"
+	set desc = "Moult into a new form." //hide if an adult?
+	set category = "Grue"
+	if(!alert(src,"Would you like to moult? You will become a vulnerable and immobile chrysalis during the process.",,"Moult","Cancel") == "Moult")
+		return
+	if (lifestage<3)
+		if (shadowpower<moultcost)
+			to_chat(src, "<span class='notice'>You need to bask in shadow more first. ([moultcost] shadow power required)</span>")
+			return
+		else if (!isturf(loc))
+			to_chat(src, "<span class='notice'>You need more room to moult.</span>")
+			return
+		else if (stat==UNCONSCIOUS)
+			to_chat(src, "<span class='notice'>You must be awake to moult.</span>")
+			return
+		else if (busy)
+			to_chat(src, "<span class='notice'>You are already doing something.</span>")
+			return
+		else
+			start_moult()
+
+	else
+		to_chat(src, "<span class='notice'>You are already fully mature.</span>")
+
+/mob/living/simple_animal/hostile/grue/proc/start_moult()
+	if(stat==CONSCIOUS&&shadowpower>=moultcost&&!ismoulting&&lifestage<3)
+		shadowpower-=moultcost
+		lifestage++
+		to_chat(src, "<span class='notice'>You begin moulting.</span>")
+		visible_message("<span class='warning'>\The [src] morphs into a chrysalis...</span>")
+		stat=UNCONSCIOUS //go unconscious while moulting
+		ismoulting=1
+		moulttimer=moulttime//reset moulting timer
+		plane = MOB_PLANE //In case grue moulted while hiding
+		var/tempHealth=health/maxHealth //to scale health level
+		if(lifestage==2)
+			desc = "A small grue chrysalis."
+			name = "grue chrysalis"
+			icon_state = "moult1"
+			icon_living = "moult1"
+			icon_dead = "moult1"
+			maxHealth=25 //vulnerable while moulting
+		else if(lifestage==3)
+			desc = "A grue chrysalis."
+			name = "grue chrysalis"
+			icon_state = "moult2"
+			icon_living = "moult2"
+			icon_dead = "moult2"
+			maxHealth=50 //vulnerable while moulting
+		health=tempHealth*maxHealth //keep same health percentage
+	else
```

Comment:
useless, remove

Reply:
Do you mean that health automatically scales when the maxHealth changes?